libs/hook-dapp-lib/src/utils.ts

```diff
@@ -13,13 +16,27 @@ export function matchHooksToDapps(hooks: CowHook[], dapps: HookDappBase[]): Hook
       acc[dapp.id] = dapp
       return acc
     },
-    {} as Record<string, HookDappBase>,
+    {} as Record<string, HookDappBase | undefined>,
   )
 
-  return hooks.map((hook) => ({
-    hook,
-    dapp: dappsMap[hook.callData.slice(-HOOK_DAPP_ID_LENGTH)] || null,
-  }))
+  return hooks.map((hook) => {
+    const dapp = dappsMap[hook.callData.slice(-HOOK_DAPP_ID_LENGTH)]
+
+    /**
+     * Permit token is a special case, as it's not a dapp, but a hook
+     */
+    if (!dapp && hook.callData.startsWith(PERMIT_SELECTOR)) {
+      return {
+        hook,
+        dapp: hookDappsRegistry.PERMIT_TOKEN as HookDappBase,
+      }
+    }
```

Comment:
This will change, but also, is it safe to filter by the permit selector?
I recall I was using it before (https://github.com/cowprotocol/cowswap/pull/3067/files/9fd0b91f7b63e481fc0c158bcd0c75833118a728#diff-006e8cd5421907fe3fd7b723ed94f3627fee5b204f8d509ba5d15ed7a06a3df8R65) but removed on Anxo's request (https://github.com/search?q=repo%3Acowprotocol%2Fcowswap+0xd505accf&type=pullrequests).
Also, there's the DAI permit selector to consider: https://github.com/cowprotocol/cowswap/blob/68566d882f0c0bebde93b8228645a59d76c2120d/libs/permit-utils/src/lib/checkIsCallDataAValidPermit.ts#L1

Reply:
I think it's ok.
Yes, there might be another (not ERC-20) contract with a method with the same name.
But probability of that this contract will be used as a hook in CoW Swap is about 0.00000000001% :)

Reply:
Got rid of it [here](https://github.com/cowprotocol/cowswap/pull/4938)